packages/transformers/js/src/visitors/dependencies.js

```diff
@@ -291,11 +306,18 @@ function addDependency(
   });
 }
 
-function addURLDependency(asset, node, opts = {}) {
-  node.value = asset.addURLDependency(node.value, {
+function addWorkerDependency(asset, node, opts = {}) {
+  let url = node.value;
+  asset.addURLDependency(url, {
     loc: node.loc && createDependencyLocation(node.loc.start, node.value, 0, 1),
     ...opts,
   });
+  morph(
+    node,
+    types.callExpression(types.identifier('require'), [
```

Comment:
@devongovett I'm concerned about the impact of resolving this dependency as a url dependency, but transforming the code to use `require`, which normally doesn't create url dependencies.

In the concatenation code path in the JS packager, we load dependencies based on a mapping from module specifier to dependency id. It's possible that a specifier here for a url dependency could collide with one from a a regular `require`, e.g. `require('foo.js')` (which should resolve to a node module package called `foo.js`), and `new Worker('foo.js')`, which should resolve as a url to a relative file named `foo.js`.

Maybe we could make changes both here and to the packager to load url dependencies differently, such as rewriting `require` to something else?

Reply:
`require('foo.js')` and `new Worker('foo.js')` should create separate dependencies since they have different environments. The JSRuntime will only process the worker one and replace its resolution, so they should end up resolving to different assets in the end. Not sure I see the problem?

Reply:
Though they'll create separate dependencies, they'll share the same `moduleSpecifier`, and can clash here:

https://github.com/parcel-bundler/parcel/blob/287ac6397ade28595d56ac73971edb832bda8ff7/packages/packagers/js/src/JSPackager.js#L110

Reply:
@wbinnssmith Can you provide a little more explanation why transforming the code to use `require` is necessary? It feels a bit hacky to me.

>Maybe we could make changes both here and to the packager to load url dependencies differently, such as rewriting require to something else?

I think I might prefer this as it would be more explicit. One downside might be that it would be something nonstandard that multiple different plugin types have to understand.

Reply:
> why transforming the code to use require is necessary? 

We need to be able to create and run runtime code. Right now `require` is the only way to do this.

Reply:
> Though they'll create separate dependencies, they'll share the same moduleSpecifier

Ah. True. Hmm, seems like we'd have to somehow use more than the module specifier as the key to dependencies...